detox/android/detox/src/main/java/com/wix/detox/LaunchIntentsFactory.java

```diff
@@ -0,0 +1,84 @@
+package com.wix.detox;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+
+class LaunchIntentsFactory {
```

Comment:
Extracted from `Detox` (thank god), with `intentWithNotificationData` as the new method to handle this use case specificially